osu.Game.Rulesets.Osu/Mods/OsuModRandom.cs

```diff
@@ -0,0 +1,364 @@
+            var relativeRotationDistance = 0f;
+            var playfieldMiddle = Vector2.Divide(OsuPlayfield.BASE_SIZE, 2);
+
+            if (prevPosChanged.X < playfieldMiddle.X)
+            {
+                relativeRotationDistance = Math.Max(
+                    (border_distance_x - prevPosChanged.X) / border_distance_x,
+                    relativeRotationDistance
+                );
+            }
+            else
+            {
+                relativeRotationDistance = Math.Max(
+                    (prevPosChanged.X - (OsuPlayfield.BASE_SIZE.X - border_distance_x)) / border_distance_x,
+                    relativeRotationDistance
+                );
+            }
+
+            if (prevPosChanged.Y < playfieldMiddle.Y)
+            {
+                relativeRotationDistance = Math.Max(
+                    (border_distance_y - prevPosChanged.Y) / border_distance_y,
+                    relativeRotationDistance
+                );
+            }
+            else
+            {
+                relativeRotationDistance = Math.Max(
+                    (prevPosChanged.Y - (OsuPlayfield.BASE_SIZE.Y - border_distance_y)) / border_distance_y,
+                    relativeRotationDistance
+                );
+            }
+
+            return rotateVectorTowardsVector(
+                posRelativeToPrev,
+                Vector2.Subtract(playfieldMiddle, prevPosChanged),
+                relativeRotationDistance
+            );
+        }
+
+        /// <summary>
+        /// Rotates vector "initial" towards vector "destinantion"
+        /// </summary>
+        /// <param name="initial">Vector to rotate to "destination"</param>
+        /// <param name="destination">Vector "initial" should be rotated to</param>
+        /// <param name="relativeDistance">The angle the vector should be rotated relative to the difference between the angles of the the two vectors.</param>
+        /// <returns>Resulting vector</returns>
+        private Vector2 rotateVectorTowardsVector(Vector2 initial, Vector2 destination, float relativeDistance)
+        {
+            var initialAngleRad = Math.Atan2(initial.Y, initial.X);
+            var destAngleRad = Math.Atan2(destination.Y, destination.X);
+
+            // Divide by 2 to limit the max. angle to 90°
+            // (90° is enough to prevent the hit objects from leaving the playfield)
+            relativeDistance /= 2;
+
+            var diff = destAngleRad - initialAngleRad;
+
+            while (diff < -Math.PI)
+            {
+                diff += 2 * Math.PI;
+            }
+
+            while (diff > Math.PI)
+            {
+                diff -= 2 * Math.PI;
+            }
+
+            var finalAngle = 0d;
+
+            if (diff > 0)
+            {
+                finalAngle = initialAngleRad + relativeDistance * diff;
+            }
+            else if (diff < 0)
+            {
+                finalAngle = initialAngleRad + relativeDistance * diff;
+            }
```

Comment:
I don't get this. Why is this `finalAngle` calculation not done unconditionally?